capstone/src/main/webapp/oAuth.js

```diff
@@ -0,0 +1,80 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Client ID from the Developer Console 
+const CLIENT_ID = 'CLIENT_ID';
+
+const AUTHORIZE_BUTTON = document.getElementById('authorize_button');
+const SIGNOUT_BUTTON = document.getElementById('signout_button');
+const SUBMIT_BUTTON = document.getElementById('submit_button');
+
+/**
+  *  On load, called to load the auth2 library and API client library.
+  */
+function handleClientLoad() {
+  gapi.load('client:auth2', initClient);
+}
+
+/**
+  *  Initializes the API client library and sets up sign-in state
+  *  listeners.
+  */
+function initClient() {
+  gapi.client.init({
+    apiKey: API_KEY,
+    clientId: CLIENT_ID,
+    discoveryDocs: DISCOVERY_DOCS,
+    scope: SCOPES
+  }).then(function () {
+    // Listen for sign-in state changes.
+    gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
```

Comment:
Just curious if you have the webpage open in 2 tabs and sign out in 1 of them, does the other tab get updated too to reflect that you are now signed out?  If so that's pretty cool (if not no worries, it's a hard problem that you don't have to address in this PR).

Reply:
I just tested this, and it looks like both pages update. As you said, pretty cool!